### PR TITLE
Adding log information and partition operation distributionsummary metrics.

### PR DESCRIPTION
**Pull request by zhljen**

### CODE REVIEW

File: metacat-common-server/src/main/java/com/netflix/metacat/common/server/monitoring/Metrics.java [hunk 1]
```diff
@@ -120,7 +126,8 @@
     enum Type {
         counter,
         gauge,
-        timer
+        timer,
+        distributionSummary
```
Comment (tgianos) [inline metacat-common-server/src/main/java/com/netflix/metacat/common/server/monitoring/Metrics.java]: What does this create?

File: metacat-main/src/main/java/com/netflix/metacat/main/services/impl/PartitionServiceImpl.java [hunk 2]
```diff
@@ -111,29 +114,35 @@ public PartitionServiceImpl(
...
+        final boolean includeUserDataMetadata,
+        final boolean includePartitionDetails
     ) {
         if (Strings.isNullOrEmpty(filter)
-                && (pageable == null || !pageable.isPageable())
-                && (partitionNames == null || partitionNames.isEmpty())
-                && config.getQualifiedNamesToThrowErrorWhenNoFilterOnListPartitions().contains(name)) {
+            && (pageable == null || !pageable.isPageable())
+            && (partitionNames == null || partitionNames.isEmpty())
+            && config.getQualifiedNamesToThrowErrorWhenNoFilterOnListPartitions().contains(name)) {
```
Comment (tgianos) [inline metacat-main/src/main/java/com/netflix/metacat/main/services/impl/PartitionServiceImpl.java]: not really your review here but man is that method name horrific

File: metacat-main/src/main/java/com/netflix/metacat/main/services/impl/PartitionServiceImpl.java [hunk 3]
```diff
@@ -156,27 +165,29 @@ public PartitionServiceImpl(
                 uris.add(partitionDto.getDataUri());
             });
 
-            registry.gauge(this.partitionGetCountId.withTags(new HashMap<>(name.parts())), result.size());
+            registry.gauge(this.partitionGetGaugeId.withTags(new HashMap<>(name.parts())), result.size());
```
Comment (tgianos) [inline metacat-main/src/main/java/com/netflix/metacat/main/services/impl/PartitionServiceImpl.java]: Here I would use `Maps.newHashMap(name.parts())` for convenience of not having "new" in the code

File: metacat-main/src/main/java/com/netflix/metacat/main/services/impl/PartitionServiceImpl.java [hunk 4]
```diff
@@ -156,27 +165,29 @@ public PartitionServiceImpl(
...
+            registry.distributionSummary(
+                this.partitionGetDistSummary.withTags(new HashMap<>(name.parts()))).record(result.size());
```
Comment (tgianos) [inline metacat-main/src/main/java/com/netflix/metacat/main/services/impl/PartitionServiceImpl.java]: same as previous

File: metacat-common-server/src/main/java/com/netflix/metacat/common/server/monitoring/Metrics.java (hunk 1 above)
Comment (zhljen) [inline metacat-common-server/src/main/java/com/netflix/metacat/common/server/monitoring/Metrics.java]: To track the historical distribution of events ( here is for the partition operations )

**Pull request merged**